test/unit/specs/core/createOptIn.spec.js

```diff
@@ -60,77 +59,64 @@ describe("createOptIn", () => {
 
       expect(optIn.isOptedIn()).toBe(false);
       optIn.whenOptedIn().then(optedInSpy);
-      return flushPromises().then(() => {
+      return flushPromiseChains().then(() => {
         expect(optedInSpy).not.toHaveBeenCalled();
       });
     });
 
     it("considers the user opted in after the user opts in", () => {
-      const optedInSpy = jasmine.createSpy();
       optIn.enable(cookie);
-      optIn.whenOptedIn().then(optedInSpy);
+      const whenOptedInPromise = optIn.whenOptedIn();
       optIn.setPurposes("all");
 
       expect(cookie.set).toHaveBeenCalledWith("optIn", "all");
       expect(optIn.isOptedIn()).toBe(true);
-      return flushPromises().then(() => {
-        expect(optedInSpy).toHaveBeenCalled();
-      });
+      return expectAsync(whenOptedInPromise).toBeResolved();
     });
 
     it("considers the user opted out after the user opts out", () => {
-      const optedOutSpy = jasmine.createSpy();
       optIn.enable(cookie);
-      optIn.whenOptedIn().catch(optedOutSpy);
+      const whenOptedInPromise = optIn.whenOptedIn();
       optIn.setPurposes("none");
 
       expect(cookie.set).toHaveBeenCalledWith("optIn", "none");
       expect(optIn.isOptedIn()).toBe(false);
-      return flushPromises().then(() => {
-        expect(optedOutSpy).toHaveBeenCalledWith(jasmine.any(Error));
-      });
+      return expectAsync(whenOptedInPromise).toBeRejectedWith(
+        jasmine.any(Error)
+      );
```

Comment:
For these cases where we're waiting for a promise to resolve or reject, we can take advantage of `expectsAsync` instead of manually flushing the promise chains.